openmc/mesh.py

```diff
@@ -809,3 +809,502 @@ def from_xml_element(cls, elem):
         mesh = cls(filename, mesh_id)
 
         return mesh
+
+  
+class WeightWindowMesh:
+    """A three dimension Weight Window Mesh class
+
+    Parameters
+    ----------
+    mesh_id : int
+        Unique identifier for the mesh
+    name : str
+        Name of the mesh
+
+    Attributes
+    ----------
+    id : int
+        Unique identifier for the mesh
+    name : str
+        Name of the mesh
+    type : int
+        weight window input file type	
+		
+	--- mesh cell boundary ---
+	origin : float
+	    lower-left coordinates for mesh
+	xmesh : float
+	    the locations of the coarse meshes in the X direction
+	xints : float
+	    the number of fine meshes within corresponding coars meshes in the X direction
+	ymesh : float
+	    the locations of the coarse meshes in the Y direction
+	yints : float
+	    the number of fine meshes within corresponding coars meshes in the Y direction
+	zmesh : float
+	    the locations of the coarse meshes in the Z direction
+	zints : float
+	    the number of fine meshes within corresponding coars meshes in the Z direction
+	--- mesh cell boundary ---
```

Comment:
What I was proposing before was that this class should just keep a reference to an instance of `openmc.RectilinearMesh` rather than defining the mesh grid here directly. So creating this class would look like:
```Python
mesh = openmc.RectilinearMesh()
mesh.x_grid = ...
mesh.y_grid = ...
mesh.z_grid = ...

ww_mesh = openmc.WeightWindowMesh(mesh)  # <-- note that we just pass the mesh created above
ww_mesh.n_energy_group = ...
...
```
This allows us to reuse all the infrastructure for rectilinear meshes (e.g., reading XML created) rather than having to duplicate it for this class specifically.

Reply:
But in the created XML file, these options (origin, xmesh, xints, ...) are needed and can't be created by the rectilinear meshes class. So I'm not sure if using the rectilinear meshes could be simpler.